src/adaptor/mod.rs

```diff
@@ -0,0 +1,21 @@
+mod ini;
+
+use std::collections::HashMap;
+use std::io::{Read, Write};
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum Config {
```

Comment:
I also don't understand why we would want to define our own type here. `serde_json::Value` has a lot of convenience methods already etc. The only difference from this type is that it also has a `Null` variant.